### DISCUSSION TITLE
[Feature Request] Have $__interval and $__interval_ms global variables available to Panel Title Field

### DISCUSSION THREAD

**Discussion by andrewmcgilvray** [grafana/grafana#36990] [Dashboards, Panels & Visualizations]
The $__interval and $__interval_ms global variables don't seem to be available to the panel Title field. They do work in the metric query field. I am using Grafana 4.2.0 on linux and the graphite datasource.

**Comment by steverweber** [comment]
related: https://github.com/grafana/grafana/issues/3940

**Comment by steverweber** [comment]
@marefr 
We hit this issue too often. It would be vary helpful if our metric titles and axes labels could include the $__interval value because without it users dont know if something is for example requests per second or minute or day...

setting a static value to lets say 1 minute makes it problematic when zooming out and viewing data over lets say one year. 

If the fix is simple the the impact would be vary helpful for us.

**Comment by steverweber** [comment]
currently what does work as a somewhat workaround is using an alias on the metric name:
```
SELECT derivative(mean("bytes_sent"), $__interval) AS "bytes sent per $__interval", derivative(mean("bytes_recv"), $__interval) AS "bytes sent per $__interval" FROM "net" WHERE ("host" = 'hpc-co1.math.private.xxx.ca') AND $timeFilter GROUP BY time($__interval) fill(null)
```

**Comment by marefr** [comment]
This is not something we've planned to work on now or short term. Contributions are welcome if you feel comfortable with that - we'll try to support you as best we can.

**Comment by silashansen** [comment]
Is there a workaround? 
It's of critical importance to quickly be able to tell if our system error-rate has been above threshold for 10 seconds or for 10 minutes, without having to hover over two adjacent points on the graph to see what the time diff is.

**Comment by Vanderdecken** [comment]
> Is there a workaround?
> It's of critical importance to quickly be able to tell if our system error-rate has been above threshold for 10 seconds or for 10 minutes, without having to hover over two adjacent points on the graph to see what the time diff is.

@silashansen You can use $__interval as a variable in your series name (in the ALIAS BY field), and then enable the Legend so you can see it statically next to the graph data. But this does require showing the legend/hovering, and it repeats for every series rather than just displaying it once in the panel title. Closest workaround I've found.

**Comment by beingyash** [comment]
Any updates on the status?

**Comment by alexanmtz** [comment]
I'm working on that @beingyash 

Issue on Gitpay: https://gitpay.me/#/task/323

**Comment by alexanmtz** [comment]
Hello everyone, I'm trying to figure out how it should work, and it would be something like here?
https://play.grafana.org/d/000000056/graphite-templated-nested?tab=advanced&orgId=1&fullscreen&edit&panelId=2

In this case, the value of `$interval` comes from repeating variables, so it would be available by default, without the need to register any variable like in this example?

So in the following MySQL Datasource, It would be possible to use `$__interval` and render on the panel title?
![pageshot of 'Dash1 - Grafana' @ 2020-03-17-1309'28](https://user-images.githubusercontent.com/88840/76855433-6be48700-6851-11ea-92aa-b9e454ce3a49.png)

can you help me with this @marefr?

**Comment by j0hnsmith** [comment]
This is a workaround to add a hidden variable that essentially copies `$__interval` to `$interval` which is then available in panel titles etc etc

```
...
   "templating": {
     "list": [
      {
        "allFormat": "glob",
        "auto": true,
        "auto_count": 20,
        "auto_min": "10s",
        "current": {
          "selected": false,
          "text": "auto",
          "value": "$__auto_interval_interval"
        },
        "datasource": null,
        "hide": 2,
        "includeAll": false,
        "label": "",
        "name": "interval",
        "options": [
          {
            "selected": true,
            "text": "auto",
            "value": "$__auto_interval_interval"
          },
          {
            "selected": false,
            "text": "1m",
            "value": "1m"
          },
          {
            "selected": false,
            "text": "2m",
            "value": "2m"
          }
        ],
        "query": "1m, 2m",
        "queryValue": "",
        "refresh": 1,
        "refresh_on_load": false,
        "skipUrlSync": false,
        "type": "interval"
      },
...
```

**Comment by ivonutar** [comment]
> This is a workaround to add a hidden variable that essentially copies `$__interval` to `$interval` which is then available in panel titles etc etc
> 
> ```
> ...
>    "templating": {
>      "list": [
>       {
>         "allFormat": "glob",
>         "auto": true,
>         "auto_count": 20,
>         "auto_min": "10s",
>         "current": {
>           "selected": false,
>           "text": "auto",
>           "value": "$__auto_interval_interval"
>         },
>         "datasource": null,
>         "hide": 2,
>         "includeAll": false,
>         "label": "",
>         "name": "interval",
>         "options": [
>           {
>             "selected": true,
>             "text": "auto",
>             "value": "$__auto_interval_interval"
>           },
>           {
>             "selected": false,
>             "text": "1m",
>             "value": "1m"
>           },
>           {
>             "selected": false,
>             "text": "2m",
>             "value": "2m"
>           }
>         ],
>         "query": "1m, 2m",
>         "queryValue": "",
>         "refresh": 1,
>         "refresh_on_load": false,
>         "skipUrlSync": false,
>         "type": "interval"
>       },
> ...
> ```

Hmm, this only shows me "1m" without considering the real interval value.

**Comment by j0hnsmith** [comment]
I tried it and it worked but then we upgraded to the latest grafana and I think it stopped working.

**Comment by w4rgrum** [comment]
A crappy workaround would be to fake the value using a dummy Prometheus query with a label_replace.
For that you simply need to create your "range" variable as a Query type on your Prometheus datasource with the following expression:
`query_result(label_replace(vector(1),"range","$__range","",""))`

![image](https://user-images.githubusercontent.com/54102154/102340289-a329c500-3f96-11eb-9be7-82dbf45d5162.png)

**Comment by bcohee** [comment]
> create your "range" variable as a Query type on your Prometheus datasource with the following expression:
> `query_result(label_replace(vector(1),"range","$__range","",""))`

Thanks for the workaround @w4rgrum 

A copy paste of the Regex:

`/^.*range="([^"]+)".*$/`

**Comment by justinTM** [comment]
@marefr ? this is a really weird and simple bug. all other variables work correctly except these specific ones. very strange.

**Comment by B3DTech** [comment]
How is this not a thing yet?  It's something simple.  I want users to see the panel title as "Requests over x" where X is the range set in the dashboard.  It doesn't seem hard, but has been an issue for years.

The workarounds seem to work for prometheus, but not other data sources, like Zabbix.

**Comment by pgassmann** [comment]
Up to Grafana 8.2, the `$__interval` variable could be used in the legend. That was a workaround, but with 8.3, this no longer works.

Especially with `count_over_time` (loki) or `sum_over_time` queries, this is necessary context information.

**Comment by gatlee21** [comment]
Any updates as to where this feature is at? This was opened in 2017 and it's now March of 2022. After 5 years it still seems relevant to the community.

**Reply by pgassmann** [reply]
The feature is just released with Grafana 8.4. I was also surprised to find it in the release notes without an update here.
https://grafana.com/blog/2022/02/17/grafana-8.4-release-new-panels-better-query-caching-increased-security-accessibility-features-and-more/

**Comment by ee-github** [comment]
But no love for $__rate_interval ?

**Reply by fangxlmr** [reply]
+1

**Comment by fangxlmr** [comment]
So any updates now?

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This feature seems to be partially resolved, so I am going to close this broader discussion. If there is an aspect of this feature that was not addressed, please create a new issue detailing what you'd like to see done, and it will be triaged and prioritized along with the team's other work.
